e2e/test/scenarios/dashboard-cards/dashboard-drill.cy.spec.js

```diff
@@ -274,42 +274,42 @@ describe("scenarios > dashboard > dashboard drill", () => {
       });
   });
 
-  it(
-    "should open the same dashboard when a custom URL click behavior points to the same dashboard (metabase#22702)",
-    { tags: "@flaky" },
-    () => {
-      createDashboardWithQuestion({}, dashboardId => {
-        visitDashboard(dashboardId);
-        cy.icon("pencil").click();
-        showDashboardCardActions();
-        cy.findByTestId("dashboardcard-actions-panel").within(() => {
-          cy.icon("click").click();
-        });
+  it("should open the same dashboard when a custom URL click behavior points to the same dashboard (metabase#22702)", () => {
+    createDashboardWithQuestion({}, dashboardId => {
+      visitDashboard(dashboardId);
+      cy.icon("pencil").click();
+      showDashboardCardActions();
+      cy.findByTestId("dashboardcard-actions-panel").within(() => {
+        cy.icon("click").click();
+      });
 
-        cy.findByText("On-click behavior for each column")
-          .parent()
-          .parent()
-          .within(() => cy.findByText("MY_NUMBER").click());
-        cy.findByText("Go to a custom destination").click();
-        cy.findByText("URL").click();
-
-        modal().within(() => {
-          cy.get("input")
-            .first()
-            .type(`/dashboard/${dashboardId}?my_param=Aaron Hand`);
-          cy.get("input").last().type("Click behavior");
-          cy.findByText("Done").click();
-        });
+      cy.findByText("On-click behavior for each column")
+        .parent()
+        .parent()
+        .within(() => cy.findByText("MY_NUMBER").click());
+      cy.findByText("Go to a custom destination").click();
+      cy.findByText("URL").click();
 
-        cy.findByText("Save").click();
+      modal().within(() => {
+        cy.get("input")
+          .first()
+          .type(`/dashboard/${dashboardId}?my_param=Aaron Hand`);
+        cy.get("input").last().type("Click behavior");
+        cy.findByText("Done").click();
+      });
 
-        cy.findByText("Click behavior").click();
+      cy.intercept("GET", "/api/dashboard/*").as("dashboard");
 
-        cy.location("pathname").should("eq", `/dashboard/${dashboardId}`);
-        cy.location("search").should("eq", "?my_param=Aaron%20Hand");
-      });
-    },
-  );
+      cy.findByText("Save").click();
+
+      cy.wait("@dashboard");
```

Comment:
Was the absence of this the actual reason for flakiness?

Reply:
@oleggromov Yes, we didn't wait for the setting to be applied and used an old value